src/main/java/operations/OperationContext.java

```diff
@@ -0,0 +1,31 @@
+package operations;
+
+import dto.PositionDto;
+import exceptions.NoValidOperationException;
+import interfaces.IOperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.Position;
+
+public class OperationContext {
+    private static final String BUY = "b";
+    private static final String RETURN = "r";
+    private static final String SUPPLY = "s";
+
+    public boolean operateSwitcher(List<PositionDto> positionDtos) {
+        Map<String, IOperationStrategy> operations = new HashMap<>();
+        operations.put(BUY, new BuyOperation());
+        operations.put(RETURN, new ReturnOperation());
+        operations.put(SUPPLY, new SupplyOperation());
```

Comment:
this code is run each time we call `opearateSwitcher()`
let's move it to static initialization block so it could be executed only once

and make `operations` static field

Reply:
done